webform_civicrm.module

```diff
@@ -1042,3 +1042,20 @@ function webform_civicrm_civicrm_pre($op, $objectName, $id, &$params) {
     }
   }
 }
+
+/**
+ * Implements hook_civicrm_alterPaymentProcessorParams().
+ *
+ * Legacy handling for paypal.
+ * We use it to override the return url so that the user gets redirected to the right place from paypal.
+ *
+ * Remove when dropping support for CiviCRM 4.6 and below.
+ */
+function webform_civicrm_civicrm_alterPaymentProcessorParams($paymentObj, $rawParams, &$cookedParams) {
+  if (!empty($rawParams['webform_redirect_cancel']) && !empty($rawParams['webform_redirect_success'])
+    && !empty($cookedParams['return']) && !empty($cookedParams['cancel_return'])
+  ) {
+    $cookedParams['return'] = $rawParams['webform_redirect_success'];
+    $cookedParams['cancel_return'] = $rawParams['webform_redirect_cancel'];
+  }
```

Comment:
If this is specifically for paypal (express?) then we really need some code comments here explaining how and why this works.

Reply:
The payment processor we use and for which I need this patch is _Paypal - Website Payments Standard_